hicommonwealth/commonwealth
# Clean Up Notifications Categories

## Link to Issue
Closes: #4370 

## Description of Changes
- Adds a migration that deletes `new-chat-mention`, `entitiy-event`, `new-role-creation`, and `new-community-creation` `NotificationCategories`
- Deleted the above categories from the `NotificationCategories` object and `server-test.ts`

## Test Plan
- Load the notifications settings page.
- Turn on or off some subscriptions.
- Check your notifications dropdown.
- Generate a notification by creating a thread or comment.

## Deployment Plan

**WARNING**: #4594 MUST be merged and deployed first.

## Other Considerations

- Migration takes 22 seconds on Frack. There should be no risk of deadlocks:
  - If [#4369](https://github.com/hicommonwealth/commonwealth/issues/4369) is merged and deployed first then there will be 0 notifications and 0 NR records to delete (there are currently 4). The delete functions are mostly to cover unforeseen cases (somehow chat mention notif is created).
  - The `new-chat-mention` subscriptions cannot be updated because they do not appear on the notifications settings dashboard. Thus, they will only be subject to `SELECT` queries which don't conflict with `DELETE` queries. None of the other deleted `NotificationCategories` have any subscriptions (and therefore NR).

[1]: https://github.com/hicommonwealth/commonwealth/issues/4370#issuecomment-1633008350
